### DISCUSSION TITLE
Required radio selections

### DISCUSSION THREAD

**Discussion by Shahin-rmz** [zauberzeug/nicegui#1362] [Q&A]
### Question

Hi,
Am I wrong, or we have to make the `radio selections`/`number` and `text inputs` manually required ?

**Comment by rodja** [comment]
What do you mean with "manually required"? Can you provide more context?

**Comment by Shahin-rmz** [comment]
I made a form, and it was pretty fast and intuitive.
I want to make some fields (Radio select, number input etc) mark as required.*
could not find any method, or property in the documentation of them.
should only manually set it like that:

```python
if radio1.value==None:
  bla bla
```
is it the right way?

**Comment by rodja** [comment]
`ui.input`, `ui.number` etc have a `validate` parameter which can you can use to check if the input is not `None`. Before sending you can check the `.error` property of each element to ensure all inputs are correct. We have ongoing discussion #801 and #1242 about possible simplifications.
